### ISSUE TITLE
`api.derive.democracy.proposals()` failed to get proposals

### ISSUE THREAD

**Issue by JayJay1024** [polkadot-js/api#4896]
`const provider = new WsProvider('wss://darwinia-crab.api.onfinality.io/public-ws');`

```js
const proposals = await api.query.democracy?.publicProps();

proposals.forEach(([, ,proposer]) => {
  console.log(proposer.isEmpty, proposer.toHuman());
});
```

the `proposer` is not empty, but `proposer.isEmpty` is `true` , what could be the root cause

<img width="853" alt="image" src="https://user-images.githubusercontent.com/21007682/172141299-680309fd-248b-418a-a265-14a7489c5680.png">

https://github.com/polkadot-js/api/blob/master/packages/api-derive/src/democracy/proposals.ts#L31

**Comment by AurevoirXavier** [comment]
I think they shield the 0x0...01 address.

**Comment by HackFisher** [comment]
> I think they shield the 0x0...01 address intentionally.

Yes, but to correct, because the proposer here is 0x00000...000.

**Comment by JayJay1024** [comment]
In the above screenshot the proposer should be 5C4hrfjw9DjXZTzV3MwzrrAr9P1MJhSrvWGWqi1eSuyUpnhM and 5C4hrfjw9DjXZTzV3MwzrrAr9P1MJhSrvWGWqi1eSuyUpnhM

**Comment by jacogr** [comment]
`5C4hrfjw9DjXZTzV3MwzrrAr9P1MJhSrvWGWqi1eSuyUpnhM` is all-empty, i.e. `0x0000.0000`

You can check in the JS dev console on apps, 

`console.log(hashing.decodeAddress('5C4hrfjw9DjXZTzV3MwzrrAr9P1MJhSrvWGWqi1eSuyUpnhM'))`

The `0x00...01` is the hash component from `(u32, Hash, AccountId)`

**Comment by HackFisher** [comment]
@jacogr Maybe we should not shield 0x0...00 here, because the private key of `0x0...00` is known to public, anyone can reveal its keys and send extrinsic as `0x0...00`.

Shielding this address in UI will prevent the governance participants from discovering these malicious proposals earlier.

**Comment by JayJay1024** [comment]
> `5C4hrfjw9DjXZTzV3MwzrrAr9P1MJhSrvWGWqi1eSuyUpnhM` is all-empty, i.e. `0x0000...0000`
> 
> You can check in the JS dev console on apps,
> 
> `console.log(hashing.decodeAddress('5C4hrfjw9DjXZTzV3MwzrrAr9P1MJhSrvWGWqi1eSuyUpnhM'))`
> 
> The `0x00...01` is the hash component from `(u32, Hash, AccountId)`

Oh I see.

**Comment by AurevoirXavier** [comment]
> @jacogr Maybe we should not shield 0x0...00 here, because the private key of `0x0...00` is known to public, anyone can reveal its keys and send extrinsic as `0x0...00`.
> 
> Shielding this address in UI will prevent the governance participants from discovering these malicious proposals earlier.

But this will also make the 'bug' more public.

**Comment by HackFisher** [comment]
> But this will also make the 'bug' more public.

It is already public, and it is from crypto, not something can be hided/fixed in UI. We should have more people know this not only for malicious, making information asymmetry more severe.

**Comment by HackFisher** [comment]
Related: https://github.com/paritytech/substrate/pull/10413

**Comment by jacogr** [comment]
Closing, nothing to be done API-wise.
